cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/io/StageTrackingOutputFormat.java

```diff
@@ -39,47 +41,53 @@ public class StageTrackingOutputFormat<K, V> extends DelegatingOutputFormat<K, V
   @Override
   public RecordWriter<K, V> getRecordWriter(TaskAttemptContext context) {
     OutputFormat<K, V> delegate = getDelegate(context.getConfiguration());
+    Configuration conf = context.getConfiguration();
     try {
       // Spark already emitting bytes written metrics for file base output,
       // hence we don't want to double count
       if (delegate instanceof FileOutputFormat) {
         return new StageTrackingRecordWriter<>(delegate.getRecordWriter(context),
-          getStageName(context.getConfiguration()));
+          getStageName(conf), ExceptionUtils.getErrorDetailsProvider(conf));
```

Comment:
If it can throw an exception, its better to create the ErrorDetailsProvider outside of this try-catch. Otherwise an error instantiating it will get treated like an error writing records.